zink/src/primitives/numeric.rs

```diff
@@ -23,6 +41,108 @@ macro_rules! impl_numeric {
     };
 }
 
+// Signed types (i32, i64)
+macro_rules! impl_safe_numeric_signed {
+    ($($t:ty);* $(;)?) => {
+        $(
+            impl SafeNumeric for $t {
+                const MAX: Self = <$t>::MAX;
+                const MIN: Self = <$t>::MIN;
+
+                #[inline(always)]
+                fn safe_add(self, rhs: Self) -> Self {
+                    let result = self.wrapping_add(rhs);
+                    if (self > 0 && rhs > 0 && result < self) || 
+                       (self < 0 && rhs < 0 && result > self) {
+                        local_revert!("addition overflow");
```

Comment:
it is because we are inside of the macro `impl_safe_numeric_signed`, the context could be different in the implementation, so, basically


```suggestion
                        zink::revert!("addition overflow");
```

when we calling safe method, we expected `zink::revert!` to be called

```
//! outside of the zink crate

a.safe_add(1);

// this should expand to

zink::revert!("addition overflow")

// instead of

crate::ffi::asm::revert!("addition overflow")

// beacause `crate::ffi` means nothing in the crates out of `zink`
```